File: app/src/androidTest/kotlin/br/com/alma/meustrocados/persistencia/ContaDAOTest.java
package br.com.alma.meustrocados.persistencia;

import org.junit.Assert;
import org.junit.Test;

import br.com.alma.meustrocados.modelo.Conta;

public class ContaDAOTest extends AppDatabaseTest {

    @Test
    public void insertConta() throws Exception {
        Conta conta = new Conta(1, "Conta", "17X-#7",1);
        Conta returningValue;

        getContaDAO().insert(conta);
        returningValue = getContaDAO().getConta(conta.uid);

        Assert.assertEquals(conta.uid, returningValue.uid);
    }

    @Test
    public void deleteconta() throws Exception {
        Conta conta = new Conta(1, "Conta", "17X-#7",1);
        Conta returningValue;

        getContaDAO().insert(conta);
        getContaDAO().delete(conta);
        returningValue = getContaDAO().getConta(conta.uid);

        Assert.assertNull(returningValue);
    }
}

File: app/src/androidTest/java/br/com/alma/mybudget/persistencia/persistencia/ContaDAOTest.kt
package br.com.alma.mybudget.persistencia.persistencia

import br.com.alma.mybudget.modelo.Conta
import org.junit.Assert
import org.junit.Test
import java.lang.Exception

class ContaDAOTest : AppDatabaseTest() {
    
    @Test
    @Throws(Exception::class)
    fun insertConta() {
        val conta: Conta = Conta(1, "Conta", "17X-#7", 1)

        contaDAO?.insert(conta)
        val returningValue: Conta = contaDAO?.getConta(conta.uid)!!

        Assert.assertEquals(conta.uid, returningValue.uid)
    }

    @Test
    @Throws(Exception::class)
    fun deleteconta() {
        val conta: Conta = Conta(1, "Conta", "17X-#7", 1)

        contaDAO?.insert(conta)
        contaDAO?.delete(conta)
        val returningValue: Conta? = contaDAO?.getConta(conta.uid)

        Assert.assertNull(returningValue)
    }
}